examples/graph_animations/example3.md

```diff
@@ -0,0 +1,239 @@
+## Minimum cost path problem
+
+In this example, we shall cover visualising the minimum-cost path finding algorithm for weighted graphs. The graph shall be represented using a custom data type. However, one could use the `MetaGraph` package from `JuliaGraphs`, where extra node properties can be stored and managed.
+
+### Djikstra's Algorithm
+
+This is one of the most famous graph algorithms concieved by computer scientist Edsger W. Dijkstra in 1956. It exists in many variants i.e. finding the shortest path from a node to another node in a weighted graph or to all other nodes. In this demo we will look at the the former variant only.
+
+The algorithm proceeds as follows:
+1. Mark all nodes as unvisited and group them into an *unvisited* set.
+2. Assign an initial weight of infinity to all nodes except the initial node which gets a weight 0. Mark this node as currrent.
+3. For every unvisited neighbour of the current node, update its weight to the minimum of its current value and the new tentative weight computed. The tentative weight is computed as the sum of the parent node value and the edge weight between the two nodes.
+4. Mark the current node as visited and remove it from unvisited set.
+5. When the destination node is marked visited, stop.
+6. Otherwise, mark the node from the nieghbouring set with the minimum weight as current and repeat step 3.
+
+### Demo
+
+**Graph creation**
+
+Create an example graph using a custom data type. For simplicity, the graph is assumed to be connected.
+```julia
+# The weights are used later in the algorithm. Storing it here avoids requiring any additional data structures. The neighbor list elements are read as (node_id, edge_weight)
+g = [Dict(:weight=>0, :neighbors=>[(2, 3), (3, 5)]),
+     Dict(:weight=>1000, :neighbors=>[(4, 1), (5, 2)]),
+     Dict(:weight=>1000, :neighbors=>[(5, 1), (4, 2)]),
+     Dict(:weight=>1000, :neighbors=>[(6, 1)]),
+     Dict(:weight=>1000, :neighbors=>[(4, 3)]),
+     Dict(:weight=>1000, :neighbors=>[])]
+```
+
+(*Visualisation coming up*)
+
+**Register graph in Javis**
+```julia
+wg = GraphAnimation(g, false, 300, 300, O; 
+                    node_attribute_fn=(g, node, attr) -> n(g, node, attr),
```

Comment:
unclear to me what this does

Reply:
Since the type of graph used is not known in this example, to fetch node properties like `weight` from the input graph this function will be used. These weights are then mapped to a certain drawing property like `radius`.

Reply:
I was wondering why this doesn't return the normal inf instead of the string as well but that's good to know.